scripts/functions.sh

```diff
@@ -205,9 +206,87 @@ function setup {
+    for option in $options; do
+        key="${option%%=*}"
+        value="${option#*=}"
+        case "$key" in
+            --enable_pca) # Enable packet capture agent
+                if [[ "$value" == "true" || "$value" == "false" ]]; then
+                  edit_manifest "pca_enable" "$value" "$manifest"
+                else
+                  echo "invalid value for --pca_enable"
+                fi
+                ;;
+            --direction) # Configure filter direction
+                if [[ "$value" == "Ingress" || "$value" == "Egress" ]]; then
+                  edit_manifest "filter_direction" "$value" "$manifest"
+                else
+                  echo "invalid value for --direction"
+                fi
+                ;;
+            --cidr) # Configure filter CIDR
+                edit_manifest "filter_cidr" "$value" "$manifest"
+                ;;
+            --protocol) # Configure filter protocol
+                if [[ "$value" == "TCP" || "$value" == "UDP" || "$value" == "SCTP" || "$value" == "ICMP" || "$value" == "ICMPv6" ]]; then
+                  edit_manifest "filter_protocol" "$value" "$manifest"
+                else
+                  echo "invalid value for --protocol"
+                fi
+                ;;
+            --sport) # Configure filter source port
+                edit_manifest "filter_sport" "$value" "$manifest"
+                ;;
+            --dport) # Configure filter destination port
+                edit_manifest "filter_dport" "$value" "$manifest"
+                ;;
+            --port) # Configure filter port
+                edit_manifest "filter_port" "$value" "$manifest"
+                ;;
+            --sport_range) # Configure filter source port range
+                edit_manifest "filter_sport_range" "$value" "$manifest"
+                ;;
+            --dport_range) # Configure filter destination port range
+                edit_manifest "filter_dport_range" "$value" "$manifest"
+                ;;
+            --port_range) # Configure filter port range
+                edit_manifest "filter_port_range" "$value" "$manifest"
+                ;;
+            --icmp_type) # ICMP type
+                edit_manifest "filter_icmp_type" "$value" "$manifest"
+                ;;
+            --icmp_code) # ICMP code
+                edit_manifest "filter_icmp_code" "$value" "$manifest"
+                ;;
+            --peer_ip) # Peer IP
+                edit_manifest "filter_peer_ip" "$value" "$manifest"
+                ;;
+            --action) # Filter action
+                if [[ "$value" == "Accept" || "$value" == "Reject" ]]; then
+                  edit_manifest "filter_action" "$value" "$manifest"
+                else
+                  echo "invalid value for --action"
+                fi
+                ;;
+            *) # Invalid option
+                echo "Invalid option: $key" >&2
+                exit 1
+                ;;
+        esac
+    done
```

Comment:
Can't we merge that loop in a single function to avoid code duplicate ?

We can add check for options not available in both flows / packets commands stopping the script

Reply:
ok will gen common function